libraries/cmake/source/lief/CMakeLists.txt

```diff
@@ -0,0 +1,388 @@
+# Copyright (c) 2014-present, The osquery authors
+#
+# This source code is licensed as defined by the LICENSE file found in the
+# root directory of this source tree.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+include(ExternalProject)
+include(CheckCCompilerFlag)
+include(CheckCXXCompilerFlag)
+
+function(liefCppMain)
+  set(library_root "${CMAKE_CURRENT_SOURCE_DIR}/src")
+  set(LIEF_ELF_SUPPORT 0)
+  set(LIEF_OAT_SUPPORT 0)
+  set(LIEF_DEX_SUPPORT 0)
+  set(LIEF_VDEX_SUPPORT 0)
+  set(LIEF_ART_SUPPORT 0)
+  set(LIEF_JSON_SUPPORT 0)
+  set(LIEF_LOGGING_SUPPORT 0)
+  set(LIEF_LOGGING_DEBUG_SUPPORT 0)
+  set(LIEF_FROZEN_ENABLED 0)
+
+  add_library(thirdparty_lief
+    "${library_root}/src/Object.cpp"
+    "${library_root}/src/Visitor.cpp"
+    "${library_root}/src/errors.cpp"
+    "${library_root}/src/exception.cpp"
+    "${library_root}/src/hash_stream.cpp"
+    "${library_root}/src/utils.cpp"
+    "${library_root}/src/Object.tcc"
+    "${library_root}/src/iostream.cpp"
+    "${library_root}/src/logging.cpp"
+    "${library_root}/src/Abstract/Binary.cpp"
+    "${library_root}/src/Abstract/EnumToString.cpp"
+    "${library_root}/src/Abstract/Function.cpp"
+    "${library_root}/src/Abstract/Header.cpp"
+    "${library_root}/src/Abstract/Parser.cpp"
+    "${library_root}/src/Abstract/Relocation.cpp"
+    "${library_root}/src/Abstract/Section.cpp"
+    "${library_root}/src/Abstract/Symbol.cpp"
+    "${library_root}/src/Abstract/hash.cpp"
+    "${library_root}/src/BinaryStream/BinaryStream.cpp"
+    "${library_root}/src/BinaryStream/Convert.cpp"
+    "${library_root}/src/BinaryStream/MemoryStream.cpp"
+    "${library_root}/src/BinaryStream/VectorStream.cpp"
+    "${library_root}/src/visitors/hash.cpp"
+    "${library_root}/src/hash_stream.hpp"
+    )
+
+  if(DEFINED PLATFORM_MACOS)
+    set(LIEF_MACHO_SUPPORT 1)
+    set(LIEF_PE_SUPPORT 0)
+    file(READ "${library_root}/include/LIEF/MachO/enums.inc"      LIEF_MACHO_ENUMS)
+    file(READ "${library_root}/include/LIEF/MachO/structures.inc" LIEF_MACHO_STRUCTURES)
+    configure_file("${library_root}/include/LIEF/MACHO/enums.hpp.in" "${library_root}/include/LIEF/MACHO/enums.hpp")
+    configure_file("${library_root}/include/LIEF/MACHO/Structures.hpp.in" "${library_root}/include/LIEF/MACHO/Structures.hpp")
```

Comment:
Do not generate these on the fly. Generate them once and store the result in `<root>/libraries/cmake/source/lief/generated/<windows or macos>/enums.hpp` etc.

Reply:
ok, thats good to know. I generated them and placed them in `<root>/libraries/cmake/source/lief/config` which is similar to the sleuthkit dependency?  I can rename `config` to `generated` if that is preferred

Reply:
Normally we use `config` if it is the direct result of a `configure` script call or if in general only contain several `preprocessor #defines` that act as a configuration for the library.
`generated` instead are other files that may contain data structures/logic. So I would say that for these enums and structures `generated` is better.

Reply:
ok that makes sense and helps.  Moved to generated folder